src/ruby/pb/grpc/health/checker.rb

```diff
@@ -48,6 +48,18 @@ def add_status(service, status)
         @status_mutex.synchronize { @statuses["#{service}"] = status }
       end
 
+      # Adds given health status for all given services
+      def set_status_for_services(status, *services)
+        services.each { |service| add_status(service, status) }
```

Comment:
Can we widen the scope `@status_mutex` in these new functions, so that all services have there status's set the new status atomically? E.g., we could replace `add_status` with `{ @statuses["#{service}"] = status }` here?

Reply:
Yes, we can replace:
```
services.each { |service| add_status(service, status) }
```
with:
```
@status_mutex.synchronize do
  services.each { |service| @statuses["#{service}"] = status }
end
```

One thing to note:
I think it would be more efficient if we introduced mutex for each service status. That way, when we operate on one service, we wouldn't lock operations on other service statuses. In other words, I suggest every service to have its own mutex.

Reply:
AFAICS setting the status of a server shouldn't be on any fast path and shouldn't need to be fast. Personally I'm more worried about causing possibly unexpected behavior, where e.g., a call to `set_status_for_services` is made, and then RPC's start seeing that new status for some servers but the old status for others; it's probably not a big deal but it might be nice if the status setting is atomic across the whole thing.